src/dev-app/datepicker/datepicker-demo.html

```diff
@@ -138,8 +138,13 @@ <h2>Datepicker with value property binding</h2>
   <mat-datepicker-toggle [for]="datePicker4"></mat-datepicker-toggle>
   <mat-form-field>
     <mat-label>Value binding</mat-label>
+    <!--  TODO Angular sets the inputs for both directives on the input element here.
+          e.g. [value] is set for MatInput and MatDatepicker.
+          MatInput#value expects a string whereas MatDatepicker#value expects a Date.
+          This breaks strict template type checking. What should we do here?
+    -->
```

Comment:
Re-opening a comment here again ([we had one in the other PR for this](https://github.com/angular/components/pull/16373#discussion_r296934263)). I'm not sure if anything has changed here. Not even sure if we have a ticket for it. I don't think we talked about it in the framework meeting yet.

Reply:
Jeremy spoke to Alex R about it. It sounds like this is a real issue and that Alex was going to investigate a fix. We should probably open a ticket in JIRA?